harvesters/src/main/java/org/fao/geonet/kernel/harvest/harvester/csw/Harvester.java

```diff
@@ -389,13 +389,6 @@ private void searchAndAlign(CswServer server, Search s, Set<String> uuids,
                 break;
             }
 
-            // Another way to escape from an infinite loop
-
-            if (returnedCount == 0) {
```

Comment:
This check was introduced as a workaround to #1429. Would the harvester still work for a server not returning `nextRecord`?